### PR TITLE
add circle_fit to poetry.lock

### PR DESCRIPTION
**Pull request by hgloeckner**

### CODE REVIEW

**Comment by hgloeckner** [comment]
closes #80

**Comment by ninarobbins** [comment]
Hey @hgloeckner, thank you for this nice pull request! I keep getting the error `TypeError: Failed to decode variable 'launch_time': cannot inherit frozen dataclass from a non-frozen one` when running the pipeline. Do you know if there is something special I should do regarding poetry?

**Comment by hgloeckner** [comment]
Hm. have you renewed your python environment? just a complete reset with conda?

**Review by tmieslinger — COMMENTED** [review]
Thanks for this major improvement on the thermodynamic quantities!
The dependencies work fine for me and the change in some moist thermo variables seems negligible (for ta, theta), and somewhat visible for theta_e (0.5% lower) and remarkable (8% lower) for RH on average for the one random test case that I ran for all sondes on 2024-09-14. I think we should merge the PR and continue in a seperate thread adding tests to the moist thermodynamics repository to check consistency between forward and backward calculations etc.

File: pydropsonde/helper/physics.py [hunk 1]
```diff
@@ -1,93 +1,70 @@
...
-    e = (
-        54.842763
-        - 6763.22 / T
-        - 4.21 * np.log(T)
-        + 0.000367 * T
-        + np.tanh(0.0415 * (T - 218.8))
-        * (53.878 - 1331.22 / T - 9.44523 * np.log(T) + 0.014025 * T)
-    )
+def theta2ta(theta, P, qv=0.0, ql=0.0, qi=0.0):
+    """Returns the temperature for an unsaturated moist fluid, given the temperature
```
Comment (ninarobbins) [inline pydropsonde/helper/physics.py:31]: given potential temperature?

File: pydropsonde/helper/__init__.py [hunk 2]
```diff
@@ -282,72 +286,30 @@ def get_si_converter_function_based_on_var(var_name):
...
-        g = np.empty(8)
-        g[0] = -2.8365744 * 10**3
-        g[1] = -6.028076559 * 10**3
-        g[2] = 1.954263612 * 10**1
-        g[3] = -2.737830188 * 10 ** (-2)
-        g[4] = 1.6261698 * 10 ** (-5)
-        g[5] = 7.0229056 * 10 ** (-10)
-        g[6] = -1.8680009 * 10 ** (-13)
-        g[7] = 2.7150305
+def q2mr(q):
```
Comment (ninarobbins) [inline pydropsonde/helper/__init__.py:289]: very minor but maybe have the description here as in the other functions?

**Review by ninarobbins — COMMENTED** [review]
It works well for me! I guess very minor comment but  some functions don't have a description yet, but this can also be done later.

**Comment by hgloeckner** [comment]
I added a description to those functions and moved them to the physics file as they fit there better^^' 

> It works well for me! I guess very minor comment but some functions don't have a description yet, but this can also be done later.

**Comment by hgloeckner** [comment]
> Thanks for this major improvement on the thermodynamic quantities! The dependencies work fine for me and the change in some moist thermo variables seems negligible (for ta, theta), and somewhat visible for theta_e (0.5% lower) and remarkable (8% lower) for RH on average for the one random test case that I ran for all sondes on 2024-09-14. I think we should merge the PR and continue in a seperate thread adding tests to the moist thermodynamics repository to check consistency between forward and backward calculations etc.

thank you so much for checking! I think, now it's also fairly easy to change the method for RH/theta calculation

**Pull request merged**